module-2/backend/week_two.md

```diff
@@ -25,21 +33,68 @@ end
 ```
 
 Now how would you find the owner of the team with an id of 4?
+Assuming a foreign key had been added to the teams table:
+Team.find(4).owner
 
 5. In a database that's holding students and teachers, what will be the relationship between students and teachers? Draw the schema diagram.
+
+Teachers have many students
+
+Teacher                  Student
+id -----------           id
+name          |          name
+class_name    |_________ teacher_id
+
+
 6. Define foreign key, primary key, and schema.
+A primary key is usually the id and Active record assigns it for you, although you can change it. A foreign key is the id for another table that you need in order to make relationships. A schema is a picture that shows all of the relationships between your tables.
+
 7. Describe the relationship between a foreign key on one table and a primary key on another table.
-8. What are the parts of an HTTP response?
+They would be the same. You take the primary key(usually the id) from one table and make it the foreign key on another table so that you can build a relationship between the two tables.
 
+8. What are the parts of an HTTP response?
+The status code, a header, and a message
 
 ### Optional Questions
 
 1. Name your five favorite ActiveRecord methods (i.e. methods your models inherit from ActiveRecord) and describe what they do.
+
+.count-returns the number of rows in your table
+.create()-makes and saves new row 
+.group().order()-groups rows by a specific attribute and then put them in order so you can call .first or .last
+.maximum() or .minimum()-gets the highest or lowest value for an attribute
+.pluck()-once you are down to a single row it can pick the value of one of the attributes out of that row
+
 2. Name your three favorite ActiveRecord rake tasks and describe what they do.
+Rake db:reset is my favorite because it does all the other rake commands with only having to type on in. Rake db:create will create the database and rake db:migrate will use the migrations to make or change tables in the database.
 3. What two columns does `t.timestamps null: false` create in our database?
+updated at and created at
 4. In a database that's holding schools and teachers, what will be the relationship between schools and teachers?
+schools have many teachers, one to many
 5. In the same database, what will you need to do to create this relationship (draw a schema diagram)?
+
+School                  Teacher
+id -----------           id
+name          |          name
+address       |_________ school_id
+
 6. Give an example of when you might want to store information besides ids on a join table.
+I don't know
```

Comment:
`invoice_items` from Black Thursday is a decent example: holds quantity and unit_price in addition to the foreign keys.